### ISSUE TITLE
NuGet won't install on VS 2015 Integrated Shell

### ISSUE THREAD

**Issue by conioh** [NuGet/Home#6674]
## Details about Problem

NuGet product used: NuGet Package Manager for Visual Studio 2015

NuGet version: 3.4.4.1321

VS version: 2015 Integrated Shell

OS version: Windows 10 x64 v1709 (16299.192)

Worked before? If so, with which NuGet version: Worked way back. Don't remember version

## Detailed repro steps so we can see the same problem

1. Get a machine (or a VM) without Visual Studio.

1. Install Visual Studio 2015 Integrated Shell or any product that has Visual Studio 2015 Integrated Shell bundled with it.

   1. Find `vs_isoshell.exe` and `vs_intshelladditional.exe` in your old downloads directory or whatever and install them in that order. Or:
   1. Install [RemObjects Elements 10](http://www.elementscompiler.com/elements/download.aspx) for Windows

1. Search the Nuget Package Manager extension from within the VS GUI. Fail

1. Download the [extension from the VS Marketplace](https://marketplace.visualstudio.com/items?itemName=NuGetTeam.NuGetPackageManagerforVisualStudio2015).

1. Try to install it.

1. Be informed that there are no compatible products install.

1. Recall that it used to work.

1. Get on GitHub and see that...

Though it's hard to follow the million renames/moves the `.vsixmanifest` files went through, [the manifests in the old repo include the `IntegratedShell` ID](https://github.com/NuGet/NuGet2/search?utf8=%E2%9C%93&q=IntegratedShell&type=).

After all the renames and moves, the current head includes 3 files:

* [`source.extension.vs14.vsixmanifest`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.VisualStudio.Client/source.extension.vs14.vsixmanifest) which declares support for Community, Pro and Express editions of VS 2015.
* [`source.extension.vs15.vsixmanifest`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.VisualStudio.Client/source.extension.vs15.vsixmanifest) which declares support for Community, Pro and Express editions of VS 2017. This doesn't makes a lot of sense since [most of the Express editions don't have a 2017 version](https://www.visualstudio.com/vs/express/) - there's only VS 2017 Express for Windows Desktop. No Web. No Windows 10. No Windows Phone.

and:

* [`source.extension.vs15.insertable.vsixmanifest`](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.VisualStudio.Client/source.extension.vs15.insertable.vsixmanifest) which declares support for `Microsoft.VisualStudio.IntegratedShell` with `Version="[15.0,16.0)"`. But there's no VS 2017 Shell! It says so everywhere. For example:

> Microsoft offers the Integrated and Isolated Shells only for Visual Studio 2015, Visual Studio 2013, Visual Studio 2012, and Visual Studio 2010. It is not available for Visual Studio 2017.

https://vspartner.com/pages/vsshells

It is known.

My guess is that somehow along the way someone just bumped up all the versions, even though it makes no sense. VS 2017 Integrated Shell is [not a thing](https://queue.acm.org/detail.cfm?id=3041020). VS 2015 Integrated Shell certainly is, but NuGet won't work with it.

Maybe my guess is wrong. I don't know what "insertable" even means and I couldn't even get the 2015 extension to build from the current head so I'm coming here.

It's worth noting that the VS2013 NuGet extension still works on VS 2013 Integrated Shell.

![image](https://user-images.githubusercontent.com/10606081/37376038-0db2cee4-272b-11e8-9bff-81fbd5db05be.png)

Please make the NuGet extensions work with VS 2015 Integrated Shell.

**Comment by conioh** [comment]
Hello?

**Comment by conioh** [comment]
If I take the https://github.com/NuGet/NuGet.Client/tree/3.4.4-rtm tag (since the version in the Visual Studio Marketplace is 3.4.4.1321) I can build it and if I apply the
```
--- a/src/NuGet.Clients/VsExtension/source.extension.vsixmanifest
+++ b/src/NuGet.Clients/VsExtension/source.extension.vsixmanifest
@@ -14,6 +14,7 @@
     <InstallationTarget Id="Microsoft.VisualStudio.VPDExpress" Version="[14.0,15.0)" />
     <InstallationTarget Id="Microsoft.VisualStudio.VSWinExpress" Version="[14.0,15.0)" />
     <InstallationTarget Id="Microsoft.VisualStudio.VSWinDesktopExpress" Version="[14.0,15.0)" />
+    <InstallationTarget Id="Microsoft.VisualStudio.IntegratedShell" Version="[14.0,15.0)" />
   </Installation>
   <Assets>
```
patch it installs on Visual Studio 2015 Integrated Shell too. Surprising.

If I download the VSIX from the marketplace, edit the manifest and repackage it, it works too (though mentioning the digital signature is obviously invalid):

![image](https://user-images.githubusercontent.com/10606081/39870605-aa62883a-546a-11e8-97b2-42d48f9662fc.png)

So, is there a reason you dropped support for the shell in 2015? Can you bring it back?

**Comment by anchal288** [comment]
any fix for this issue? i need nuget on 2015 shell to code the integration with powerBI. appreciate any help with the solution/workaround.

**Comment by conioh** [comment]
@anchal288, seems like Microsoft - once again - doesn't even bother to look at bug reports.
Last time they ignored bug reports they deleted many Terabytes if not Petabytes of users' files.
If you really need the functionality and can't use VS Community because if the license, I'd suggest doing what I described earlier.

**Comment by conioh** [comment]
Ola?

**Comment by conioh** [comment]
Lame.
